### PR TITLE
NH-76572 - Removed maven release plugin

### PR DESCRIPTION
**Pull request by jerrytfleung**
### Description of changes:
- Removed maven release plugin
- Updated Makefile
- Removed go.mod and go.sum

### Related issues #
[NH-76572](https://swicloud.atlassian.net/browse/NH-76572)

[NH-76572]: https://swicloud.atlassian.net/browse/NH-76572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,9 +3,9 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <groupId>cloud.solarwindscloud</groupId>
+  <groupId>cloud.solarwinds</groupId>
```
Comment (cleverchuk) [inline pom.xml:6]: Can we do `com` instead of `cloud` because `cloud` doesn't look like a real TLD?

File: pom.xml [hunk 2]
```diff
@@ -3,9 +3,9 @@
...
   <artifactId>apm-proto</artifactId>
-  <version>1.0.4-SNAPSHOT</version>
+  <version>0.0.0-SNAPSHOT</version>
```
Comment (cleverchuk) [inline pom.xml:8]: It looks like this should be `1.0.5-*`? How are we going backwards?

Comment (jerrytfleung) [reply pom.xml:8]: This version will always be set by github action release.yml. So this field now is a placeholder.

File: pom.xml (hunk 1 above)
Comment (jerrytfleung) [reply pom.xml]: Done

**Review by cleverchuk — APPROVED** [review]

File: go/collectorpb/go.mod
[the remark this answers is not in the document]
Comment (jerrytfleung) [reply go/collectorpb/go.mod:1]: go.mod keeps changing after each build. e.g. I encountered 
google.golang.org/protobuf v1.34.1 -> v1.34.2 after `go mod tidy` in Makefile.

For the go folder, I just need the code.

**Pull request merged**